### PR TITLE
Simplify sample pipeline input parameters

### PR DESCRIPTION
**Pull request by gmfrasca**
## The issue resolved by this Pull Request:
Resolves #annoyance

## Description of your changes:
standard_scaler and min_max_scaler were effectivly mutually exclusive, so let's implement them as such to work around the issue with false-y default input params not parsing/instantiating correctly on the kfp UI.

## Testing instructions
Deploy DSPO, create a v2 DSPA with `spec.apiServer.enableSamplePipeline` = true.  Create a run of the sample pipeline and ensure that all input parameters are populated and you can simply click 'run' without specifying anything else.  Extra credit for also setting 'standard_scaler' to false and ensuring pipeline still runs with the MinMaxScaler instead of StandardScaler

## Checklist
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by dsp-developers** [comment]
A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-611`
An OCP cluster where you are logged in as cluster admin is required.

To use this image run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines-operator.git
cd data-science-pipelines-operator/
git fetch origin pull/611/head
git checkout -b pullrequest debdd18dd298287080d3e021568ccf76c6fce2e6
oc new-project opendatahub
make deploy IMG="quay.io/opendatahub/data-science-pipelines-operator:pr-611"
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-611`

File: config/internal/apiserver/sample-pipeline/sample-pipeline.yaml.tmpl [hunk 1]
```diff
@@ -97,21 +88,21 @@ data:
                 \    col_names = [\n        'Sepal_Length', 'Sepal_Width', 'Petal_Length',\
                 \ 'Petal_Width', 'Labels'\n    ]\n    df = pd.read_csv(csv_url, names=col_names)\n\
                 \n    with open(iris_dataset.path, 'w') as f:\n        df.to_csv(f)\n\n"
-              image: quay.io/hukhan/iris-base:1
+              image: quay.io/opendatahub/ds-pipelines-sample-base:v1.0
```
Comment (gregsheremeta) [inline config/internal/apiserver/sample-pipeline/sample-pipeline.yaml.tmpl:91]: we should just use python base here

the docker rate limiting is a pain, but I also don't want to mislead people into thinking they need to use some special base image ... so feel free to point to a quay python if such a thing exists

File: config/internal/apiserver/sample-pipeline/sample-pipeline.yaml.tmpl [hunk 2]
```diff
@@ -72,16 +63,16 @@ data:
...
               - create_dataset
               command:
               - sh
               - -c
               - "\nif ! [ -x \"$(command -v pip)\" ]; then\n    python3 -m ensurepip ||\
                 \ python3 -m ensurepip --user || apt-get install python3-pip\nfi\n\nPIP_DISABLE_PIP_VERSION_CHECK=1\
-                \ python3 -m pip install --quiet     --no-warn-script-location 'kfp==2.0.1'\
-                \ && \"$0\" \"$@\"\n"
+                \ python3 -m pip install --quiet     --no-warn-script-location 'pandas==2.2.0'\
+                \ 'kfp==2.0.0-beta.13' && \"$0\" \"$@\"\n"
```
Comment (gregsheremeta) [inline config/internal/apiserver/sample-pipeline/sample-pipeline.yaml.tmpl:75]: is this compiled with an ancient kfp sdk? 2.7.0 should work

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-611`

File: config/internal/apiserver/sample-pipeline/sample-pipeline.yaml.tmpl (hunk 2 above)
Comment (gmfrasca) [reply config/internal/apiserver/sample-pipeline/sample-pipeline.yaml.tmpl]: was trying to see why metrics output artifact had magically disappeared.  turns out the pipeline in `config/internal/apiserver/sample-pipeline.yaml.tmpl` (which was compiled with the now-defunct kfp-2.0.1) was mismatched with the one in `docs/`.  This and the mismatch have now been updated to maintain sync

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-611`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-611`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-611`

File: config/internal/apiserver/sample-pipeline/sample-pipeline.yaml.tmpl (hunk 1 above)
Comment (gmfrasca) [reply config/internal/apiserver/sample-pipeline/sample-pipeline.yaml.tmpl:100]: for tracability i added a GH issue here: https://github.com/opendatahub-io/data-science-pipelines-operator/issues/612

**Review by amadhusu — APPROVED** [review]
Tackled this with extra credit by setting `standard_scaler` to false as well just to see what happens. It ran successfully but the details doesn't show that for the run.

Screenshot from 2024-03-20 18-35-48

**Review by DharmitD — APPROVED** [review]
Changes lgtm and since Achyut's also tested them out

/Approve

**Pull request merged**